odoo/service/server.py

```diff
@@ -434,6 +438,45 @@ def target():
             t.start()
             _logger.debug("cron%d started!" % i)
 
+    def async_thread(self, n):
+        from odoo.addons.base.models.ir_async import ir_async
+        conn = odoo.sql_db.db_connect('postgres')
+
+        with conn.cursor() as cr:
+            pg_conn = cr._cnx
+            cr.execute("LISTEN odoo_async")
+            cr.commit()
+
+            notified_dbs = [
+                dbname
+                for dbname, registry
+                in odoo.modules.registry.Registry.registries.items()
+                if registry.ready
+            ]
```

Comment:
The problem with this is that the registries may change over time, but the async thread will always poll the same ones...  I suggest the code below to loop over `list(odoo.modules.registry.Registry.registries)` instead.

Reply:
that notified_dbs is only to poll all databases on startup, it then get the notified db from the NOTIFY payload